src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -215,9 +215,14 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       void afterResolve(ResolvableDependencies resolvableDependencies) {}
     })
 
-    project.tasks.withType(AbstractCompile).all { AbstractCompile compile ->
-      CheckerFrameworkTaskExtension checkerExtension =
-          compile.extensions.create("checkerFramework", CheckerFrameworkTaskExtension)
+    project.tasks.withType(AbstractCompile).configureEach { AbstractCompile compile ->
+      LOG.debug('Adding checkerFramework configuration to task ' + compile.name)
+      try {
+        compile.extensions.create("checkerFramework", CheckerFrameworkTaskExtension)
+      } catch (IllegalArgumentException e) {
+        // Sometimes this happens in Gradle 6.4+; don't crash just in case.
+        LOG.info("Task " + compile.name + " already has the checkerFramework extension.")
```

Comment:
Does this happen only when `configureEach` is used? Does it happen with `.all` as well?

Frankly speaking, it looks suspicious.

Reply:
> Frankly speaking, it looks suspicious.

I agree! The docs say that `extensions.create` only throws `IllegalArgumentException` when the extension already exists, and I did some debugging to confirm that the extension was the correct class, so it seemed spurious.

Let me check again now that it uses `configureEach`.

Reply:
It looks like this happens when the Checker Framework plugin is added after the Java plugin, which some of the tests do despite warning against it in `README.md` -- I think I'll keep the check in because it doesn't seem worth crashing a build over, but add a warning.

Reply:
Oh, the real cause is that `configureProject` is called multiple times: https://github.com/kelloggm/checkerframework-gradle-plugin/blob/1dab41aea433b3836e3628a7a1dd4626fe9188fc/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy#L83-L95

It would probably make sense to refactor that, however, what do you think if an explicit `if (compile.extensions.findByName("checkerFramework")==null)` was added instead of `catch(IllegalArgumentException)` ?

Reply:
Done.